File: data/src/main/java/fup/prototype/data/di/DataModule.java
package fup.prototype.data.di;

import android.support.annotation.NonNull;

import dagger.Module;
import dagger.Provides;
import fup.prototype.data.DatabaseDao;
import fup.prototype.data.details.RepositoryDatabaseProvider;
import fup.prototype.data.search.UserDatabaseProvider;

@Module
public class DataModule {

    @Provides
    DatabaseDao provideUserDatabaseProvider(@NonNull final UserDatabaseProvider userDatabaseProvider) {
        return userDatabaseProvider;
    }

    @Provides
    DatabaseDao provideRepositoryDatabaseProvider(@NonNull final RepositoryDatabaseProvider repositoryDatabaseProvider) {
        return repositoryDatabaseProvider;
    }
}

File: data/src/main/java/com/rxdroid/data/di/DataModule.kt
package com.rxdroid.data.di

import com.rxdroid.data.details.RepositoryDatabaseProvider
import com.rxdroid.data.details.RepositoryDatabaseProviderImpl
import com.rxdroid.data.search.UserDatabaseProvider
import com.rxdroid.data.search.UserDatabaseProviderImpl
import org.koin.android.ext.koin.androidContext
import org.koin.dsl.module.module

val dataModule = module {

    single { UserDatabaseProviderImpl(androidContext()) as UserDatabaseProvider }

    single { RepositoryDatabaseProviderImpl(androidContext()) as RepositoryDatabaseProvider }

}